pkg/apis/serving/v1alpha2/kfservice_status.go

```diff
@@ -0,0 +1,121 @@
+
+package v1alpha2
+
+import (
+	"github.com/knative/pkg/apis"
+	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"k8s.io/api/core/v1"
+)
+
+// ConditionType represents a Service condition value
+const (
+	// RoutesReady is set when network configuration has completed.
+	RoutesReady apis.ConditionType = "RoutesReady"
+	// DefaultPredictorReady is set when default predictor has reported readiness.
+	DefaultPredictorReady apis.ConditionType = "DefaultPredictorReady"
+	// CanaryPredictorReady is set when canary predictor has reported readiness.
+	CanaryPredictorReady apis.ConditionType = "CanaryPredictorReady"
+)
+
+// KFService Ready condition is depending on default predictor and route readiness condition
+// canary readiness condition only present when canary is used and currently does
+// not affect KFService readiness condition.
+var conditionSet = apis.NewLivingConditionSet(
+	DefaultPredictorReady,
+	RoutesReady,
+)
+
+var _ apis.ConditionsAccessor = (*KFServiceStatus)(nil)
+
+func (ss *KFServiceStatus) InitializeConditions() {
+	conditionSet.Manage(ss).InitializeConditions()
+}
+
+// IsReady returns if the service is ready to serve the requested configuration.
+func (ss *KFServiceStatus) IsReady() bool {
+	return conditionSet.Manage(ss).IsHappy()
+}
+
+// GetCondition returns the condition by name.
+func (ss *KFServiceStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return conditionSet.Manage(ss).GetCondition(t)
+}
+
+// PropagateDefaultConfigurationStatus propagates the default Configuration status and applies its values
+// to the Service status.
+func (ss *KFServiceStatus) PropagateDefaultConfigurationStatus(defaultConfigurationStatus *knservingv1alpha1.ConfigurationStatus) {
+	ss.Default.Name = defaultConfigurationStatus.LatestCreatedRevisionName
+	configurationCondition := defaultConfigurationStatus.GetCondition(knservingv1alpha1.ConfigurationConditionReady)
+
+	switch {
+	case configurationCondition == nil:
+	case configurationCondition.Status == v1.ConditionUnknown:
+		conditionSet.Manage(ss).MarkUnknown(DefaultPredictorReady, configurationCondition.Reason, configurationCondition.Message)
+	case configurationCondition.Status == v1.ConditionTrue:
+		conditionSet.Manage(ss).MarkTrue(DefaultPredictorReady)
+	case configurationCondition.Status == v1.ConditionFalse:
+		conditionSet.Manage(ss).MarkFalse(DefaultPredictorReady, configurationCondition.Reason, configurationCondition.Message)
+	}
+}
+
+// PropagateCanaryConfigurationStatus propagates the canary Configuration status and applies its values
+// to the Service status.
+func (ss *KFServiceStatus) PropagateCanaryConfigurationStatus(canaryConfigurationStatus *knservingv1alpha1.ConfigurationStatus) {
+	// reset status if canaryConfigurationStatus is nil
+	if canaryConfigurationStatus == nil {
+		ss.Canary = StatusConfigurationSpec{}
+		conditionSet.Manage(ss).MarkUnknown(CanaryPredictorReady, "CanarySpecUnavailable", "Canary spec unavailable")
```

Comment:
this copy of the code is so sad... it misses for instance the fix just merged with #285 
can we please have a strategy that does not require duplication of all the code?
or - just bite the bullet - delete the old code and move to the new path quickly to minimize divergence?
@johnugeorge  @ellis-bigelow

Reply:
Let's close this PR and start with a new PR just renaming v1alpha1 to v1alpha2, then add v1alpha2 changes on top of that, what do you guys think ?

Reply:
Makes sense!